app/src/main/java/com/portfolio/tasky/entry/repositories/EntryRepositoryImpl.kt

```diff
@@ -0,0 +1,59 @@
+package com.portfolio.tasky.entry.repositories
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.portfolio.tasky.entry.network.EntryApiCall
+import com.portfolio.tasky.entry.models.AuthenticationRequest
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import com.portfolio.tasky.entry.models.RegisterRequest
+import com.portfolio.tasky.networking.usecases.domain.TaskyCallStatus
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import retrofit2.Response
+import javax.inject.Inject
+
+
+class EntryRepositoryImpl@Inject constructor(private val entryApiCall: EntryApiCall, private val taskyCallStatus: TaskyCallStatus) : EntryRepository  {
+    private val registrationLiveData = MutableLiveData<Boolean>()
+    private val loginLiveData = MutableLiveData<AuthenticationResponse?>()
+
+    private val coroutineExceptionHandler = CoroutineExceptionHandler{ _, throwable ->
+        throwable.printStackTrace()
+    }
+
+    override fun doRegistration(registerRequest: RegisterRequest): MutableLiveData<Boolean> {
```

Comment:
Why MutableLiveData?

Also, LiveData, Flows, etc is used when you'll get more than just 1 result
Here, you'll only get 1. Either a success with a body, or an error

Reply:
because I want to update the LiveData who has this MutableLiveData reference.

Currently I need only true or false in terms of registration, thats why used a boolean liveData. On success or failure it will update the instance of the live data that is being observed on UI.

Reply:
Right,. But here you could just return a boolean (no need for live data), and change the livedata in the ViewModel based on the result